<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Wurl</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QLabel" name="lbRequest">
       <property name="font">
        <font>
         <family>Verdana</family>
         <pointsize>9</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Request</string>
       </property>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget">
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="2">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="lbUrl">
          <property name="text">
           <string>URL: </string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="lbTimeout">
          <property name="text">
           <string>Timeout:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBox"/>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QLineEdit" name="leUrl"/>
        </item>
       </layout>
      </widget>
      <widget class="QLabel" name="lbActions">
       <property name="font">
        <font>
         <family>Verdana</family>
         <pointsize>9</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Actions</string>
       </property>
      </widget>
      <widget class="QWidget" name="horizontalLayoutWidget">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="buttonGet">
          <property name="text">
           <string>GET</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonPost">
          <property name="text">
           <string>POST</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonPut">
          <property name="text">
           <string>PUT</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbVerbs">
          <property name="currentText">
           <string>GET</string>
          </property>
          <item>
           <property name="text">
            <string>GET</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>CONNECT</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>COPY</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DELETE</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>HEAD</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MKCOL</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MOVE</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>OPTIONS</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>POST</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PUT</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PROPFIND</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TRACE</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonSend">
          <property name="text">
           <string>SEND</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Content to Send</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QGridLayout" name="gl">
           <item row="0" column="0">
            <widget class="QLabel" name="lbFile">
             <property name="text">
              <string>File:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QToolButton" name="tbBrowse">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lbContentType">
             <property name="text">
              <string>Content Type:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lbContentOptions">
             <property name="text">
              <string>Content Options:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="leContentType"/>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="leFile"/>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="hl">
             <item>
              <layout class="QVBoxLayout" name="vl">
               <item>
                <widget class="QCheckBox" name="cbBase64">
                 <property name="text">
                  <string>Base64</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cbParameterBody">
                 <property name="text">
                  <string>Parameter Body</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="pteRequestBody"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tabHeaders">
        <attribute name="title">
         <string>Headers</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QGridLayout" name="glHeaders">
           <item row="1" column="0">
            <widget class="QLabel" name="lbValue">
             <property name="text">
              <string>Value:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="buttonAddModifyHeaders">
             <property name="text">
              <string>Add / Change</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="lbName">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <spacer name="hs">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QLineEdit" name="leHeaderValue"/>
           </item>
           <item row="0" column="1" colspan="2">
            <widget class="QLineEdit" name="leHeaderName"/>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTableWidget" name="twHeaders">
           <property name="frameShape">
            <enum>QFrame::Box</enum>
           </property>
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tabParameters">
        <attribute name="title">
         <string>Parameters</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="3" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Value:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QPushButton" name="pushButton_2">
             <property name="text">
              <string>Add / Change</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="3" column="1" colspan="2">
            <widget class="QLineEdit" name="lineEdit_2"/>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QLineEdit" name="lineEdit"/>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTableWidget" name="tableWidget_2">
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Request Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QTableWidget" name="twRequestLog">
          <property name="frameShape">
           <enum>QFrame::Box</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Sunken</enum>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="gridStyle">
           <enum>Qt::NoPen</enum>
          </property>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Method</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>URL</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Content Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Data</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Response Log</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QTableWidget" name="twResponseLog">
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Code</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Headers</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Data</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Response Header</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QTableWidget" name="twResponseHeader">
          <property name="frameShape">
           <enum>QFrame::Box</enum>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="gridStyle">
           <enum>Qt::NoPen</enum>
          </property>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Field</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Value</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabResponseBody">
       <attribute name="title">
        <string>Response Body</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QPlainTextEdit" name="pteResponseBody"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>940</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_Settings">
    <property name="title">
     <string>&amp;Settings</string>
    </property>
    <addaction name="actionSettingsLoad"/>
    <addaction name="actionSettingsSave"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Settings"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionSettingsLoad">
   <property name="text">
    <string>&amp;Load</string>
   </property>
  </action>
  <action name="actionSettingsSave">
   <property name="text">
    <string>S&amp;ave</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>buttonGet</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onButtonGet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonPost</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onButtonPost()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>723</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonPut</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onButtonPut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>724</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonSend</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onButtonSend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>415</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>724</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tbBrowse</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onFileBrowse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>404</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>723</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonAddModifyHeaders</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onAddChangeHeaders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>724</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onAddChangeParameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>725</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onFileExit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>469</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettingsLoad</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSettingsLoad()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>469</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettingsSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSettingsSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>469</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onButtonGet()</slot>
  <slot>onButtonPost()</slot>
  <slot>onButtonPut()</slot>
  <slot>onButtonSend()</slot>
  <slot>onFileBrowse()</slot>
  <slot>onAddChangeHeaders()</slot>
  <slot>onAddChangeParameters()</slot>
  <slot>onFileExit()</slot>
  <slot>onSettingsLoad()</slot>
  <slot>onSettingsSave()</slot>
 </slots>
</ui>
